tech-specs/src/explore/design/dkg.md

```diff
@@ -1 +1,154 @@
+However, when should the proposer make this decision? In all likelihood it will be a combination of blocks transpired since
+the last instance completed as well as special cases (such as an instance
+failing to complete).
+
+## Announcing and distributing weight shares
+
+In order for the DKG protocol to work, it requires the contribution of 
+ \\(\frac{2}{3} \\) of the total staked weight among live validators. In 
+order to know when a subset of validators own at least \\(\frac{2}{3}\\)
+of the total weight, it is necessary for validators to announce their stake.
+
+Furthermore, the total weight will be scaled down to a fixed parameter 
+\\( W\\) (and fractional weights rounded appropriately to acheive integral
+values). After this procedure, each validator has a corresponding number of
+weight shares. 
+
+Lastly, the DKG protocol requires a canonical ordering of validators with
+shares must be agreed upon. This will be done via the consensus layer.
+The current block proposer will apportion out weight shares as well as the
+aforementioned ordering (which induces a natural partition of the weight 
+shares) to each validator that made an announcement. This is included in the
+block. Once this block is on the blockchain, the next step of the DKG can proceed.
+
+## Dealers
+In a single DKG instance, a subset of validators holding at least
+\\( \frac{2}{3} \\) of the weight shares must each act as dealers. It
+is the responsibility of a dealer to post a signed message to the blockchain
+which includes the session id (\\( \tau \\)) as well the data of the
+PVSS (we do not concern ourselves with the details here). The creation of this
+message is handled by the Ferveo library.
+
+This means that the signed message is a transaction that must be gossiped and
+eventually included on the blockchain. For this particular transaction, there
+is a special verification that must be done, otherwise the transaction should
+not be included in any proposed block. This verification is provided by the
+Ferveo library.
+
+## Aggregation
+
+At some point, enough validators (meaning their combined weight  shares
+exceeds \\( \frac{2}{3}\\) of the total) have their PVSS on the chain 
+for the session with id \\( \tau \\). At this point, the next block proposer can
+aggregate the messages into a single instance using Ferveo. This is then 
+included in the next block. Once this DKG is on the blockchain, it can be 
+used for encryption of transactions until the next DKG instance is run.
+
+When deciding to aggregate the PVSS instances into a single one, we shall 
+consider the instances ranked in terms of the weights of their dealers (in 
+descending order). The topmost dealers constituting enough weight are then 
+chosen and aggregated in their specified order.
+
+We note that enough weight may be gossiped to the current proposer and thus
+they can put the PVSS transactions and aggregation all in the same proposed 
+block.
+
```

Comment:
Note that any number of PVSS transcripts can be aggregated. So even if we don't get enough weight we can still propose one transaction which will be the partial aggregation the current PVSS transcripts.

Reply:
it's not clear to me if there is a benefit to partial aggregation (unlike the asynchronous case, where partial aggregation is a critical part of the protocol)

Mostly because in the on-chain aggregation case, only one final aggregation is needed, so I think it's hard to do better than that by splitting up the aggregation.

Reply:
I actually to to update this bit as the proposer should not receive these PVSS via gossiping but rather as part of the `VoteExtension` phase. They will aggregate as part of the `VerifyVoteExtension` phase and this will be put in the block header. Then we *should* get enough PVSS transcripts as otherwise we would not have received enough precommits for the block to be finalized anyway. The only way we would not (barring problems with the rounding of the weight shares) is that some validator sent a vote extension without a PVSS transcript. Do we want to punish such behavior?

Reply:
Ok, that sounds very reasonable (as long as the amount of data is not too big)
I don't know if punishing failure to attach a PVSS matters. I would suggest just treating it like a missing vote (which I guess counts as punishment)

Reply:
I am a bit concerned about the size of the data. Especially as you said elsewhere these transcripts are rather large. From the abci++ perspective, it is pretty clear they want these kind of extra protocols done during vote extension phases but it puts a lot of weight on our headers.

Reply:
I had been imagining that we could do fewer transcripts per block (over more blocks) but if that is not really compatible with vote extensions, then we could split transcripts in the opposite direction: everyone posts a tiny bit of their transcript per block. Although it's slightly more complex to keep track of which validators missed posting some part of their transcripts and need to continue to provide more

Reply:
What we could do is to decide to split up the posting of the PVSS transcripts across `n` blocks. Since these are ordered, we say at block `i` that the transcripts corresponding to weight shares between `n*i` and `n*(i+1)` should be submitted.